Metadata/Driver/AnnotationDriver.php

```diff
@@ -72,30 +72,38 @@ public function loadMetadataForClass(\ReflectionClass $class)
 
         foreach ($this->reader->getClassAnnotations($class) as $annot) {
             if ($annot instanceof Service) {
+                $service = array();
                 if (null === $annot->id) {
-                    $metadata->id = $this->namingStrategy->classToServiceName($className);
+                    $service['id'] = $this->namingStrategy->classToServiceName($className);
                 } else {
-                    $metadata->id = $annot->id;
+                    $service['id'] = $annot->id;
                 }
 
-                $metadata->parent = $annot->parent;
-                $metadata->public = $annot->public;
-                $metadata->scope = $annot->scope;
-                $metadata->abstract = $annot->abstract;
+                $service['parent'] = $annot->parent;
+                $service['public'] = $annot->public;
+                $service['scope'] = $annot->scope;
+                $service['abstract'] = $annot->abstract;
+
+                // remove all empty values from service definition
+                $metadata->addService(array_filter($service));
```

Comment:
I think there are more benefits to keep them (we don't have to always check if the values exist).